fastai/tabular/transform.py

```diff
@@ -13,8 +15,45 @@ def make_date(df:DataFrame, date_field:str):
     if not np.issubdtype(field_dtype, np.datetime64):
         df[date_field] = pd.to_datetime(df[date_field], infer_datetime_format=True)
 
+def cyclic_dt_feat_names(time:bool=True, add_linear:bool=False)->List[str]:
+    "Return feature names of date/time cycles as produced by `cyclic_dt_features`."
+    fs = ['cos','sin']
+    attr = [f'{r}_{f}' for r in 'weekday day_month month_year day_year'.split() for f in fs]
+    if time: attr += [f'{r}_{f}' for r in 'hour clock min sec'.split() for f in fs]
+    if add_linear: attr.append('year_lin')
+    return attr
+
+def cyclic_dt_features(d:Union[date,datetime], time:bool=True, add_linear:bool=False)->List[float]:
+    "Calculate the cos and sin of date/time cycles."
+    tt,fs = d.timetuple(), [np.cos, np.sin]
+    day_year,days_month = tt.tm_yday, calendar.monthrange(d.year, d.month)[1]
+    days_year = 366 if calendar.isleap(d.year) else 365
+    rs = d.weekday()/7, (d.day-1)/days_month, (d.month-1)/12, (day_year-1)/days_year
+    feats = [f(r * 2 * np.pi) for r in rs for f in fs]
+    if time and isinstance(d, datetime) and type(d) != date:
+        rs = tt.tm_hour/24, tt.tm_hour%12/12, tt.tm_min/60, tt.tm_sec/60
+        feats += [f(r * 2 * np.pi) for r in rs for f in fs]
+    if add_linear:
+        if type(d) == date: feats.append(d.year + rs[-1])
+        else:
+            secs_in_year = (datetime(d.year+1, 1, 1) - datetime(d.year, 1, 1)).total_seconds()
+            feats.append(d.year + ((d - datetime(d.year, 1, 1)).total_seconds() / secs_in_year))
+    return feats
+
+def add_cyclic_datepart(df:DataFrame, field_name:str, prefix:str=None, drop:bool=True, time:bool=False, **kwargs):
```

Comment:
We're trying to avoid using unnecessary kwargs now, so that users can see every argument in the documentation. Here you're only hiding add_linear in those kwargs, so I'd explicitly name it.

Reply:
Updated (I also prefer it that way).